tests/unit/test_dto/test_factory/test_integration.py

```diff
@@ -681,3 +682,54 @@ def handler() -> Response[List[User]]:
     with TestClient(app=module.app) as client:
         response = client.get("/")
         assert response.json() == [{"name": "John"}, {"name": "Jane"}]
+
+
+def test_dto_with_msgspec_and_generic_inherited_models(create_module: Callable[[str], ModuleType]) -> None:
+    module = create_module(
```

Comment:
why are you creating a module here? 

I would assume you can simply rewrite this test to use the kind of logic we have everywhere else, and have the imports top level

Reply:
I was actually going to ask on Discord. I just noticed a lot of tests in that file are written like this so I copied one and made some adjustments.

I don't know what is the motive behind that. I'd convert it to normal tests if this isn't a convention for that file.

Reply:
The reason this is done in some cases is to test behaviour with `from __future__ import annotations`, independently of the module containing the test.